drivers/temperature/adt7420/adt7420.c

```diff
@@ -205,15 +272,48 @@ void adt7420_set_operation_mode(struct adt7420_dev *dev,
 void adt7420_set_resolution(struct adt7420_dev *dev,
 			    uint8_t resolution)
 {
-	uint8_t register_value = 0;
+	uint8_t register_value[1] = { 0 };
+	uint8_t device_reg_address;
+
+	if (chip_info[dev->active_device].communication == SPI) {
+		device_reg_address = ADT7320_REG_CONFIG;
+		register_value[0]  = adt7420_get_register_value(dev, device_reg_address);
+	} else {
+		device_reg_address = ADT7420_REG_CONFIG;
+		register_value[0]  = adt7420_get_register_value(dev, device_reg_address);
+	}
+
+	register_value[0] &= ~ADT7420_CONFIG_RESOLUTION;
+	register_value[0] |= (resolution * ADT7420_CONFIG_RESOLUTION);
 
-	register_value  = adt7420_get_register_value(dev, ADT7420_REG_CONFIG);
-	register_value &= ~ADT7420_CONFIG_RESOLUTION;
-	register_value |= (resolution * ADT7420_CONFIG_RESOLUTION);
-	adt7420_set_register_value(dev, ADT7420_REG_CONFIG, register_value);
+
+	set_register_value(dev, device_reg_address, 2, register_value);
 	dev->resolution_setting = resolution;
 }
 
+/***************************************************************************//**
+ * @brief Resets the SPI or I2C inteface for the ADT7420/ADT7320
+ *
+ * @param dev        - The device structure.
+ *
+ * @return None.
+*******************************************************************************/
+void adt7420_reset_interface(struct adt7420_dev *dev)
+{
+	uint8_t data_buffer[] = { 0xFF, 0xFF, 0xFF, 0xFF };
+
+	if (chip_info[dev->active_device].communication == SPI)
+		spi_write_and_read(dev->spi_desc, data_buffer, sizeof(data_buffer));
+	else {
+		uint8_t register_address = ADT7420_REG_RESET;
+		i2c_write(dev->i2c_desc,
+			  &register_address,
+			  1,
+			  1);//no repeat start
+	}
+	dev->resolution_setting = 0;
```

Comment:
you need to check for the error codes above and if all was ok, then consider the reset successful

when you know for sure the reset was successful you probably need to set some device driver parameters to their default - which you do in the case of resolution_setting, but the question is, are there other members that need to be reset to default as well ?

Reply:
Error code check has been added. 

Regarding the other members to be reset - The only member in the **adt7420_dev** structure that may need a manual reset would be the resolution setting. Other members are the spi/i2c init parameters and the active device, which may get selected upon executing the application. There may not be a relevant default case for the same.